api/handler/acl.go

```diff
@@ -1190,6 +1190,13 @@ func aclToPolicy(acl *AccessControlPolicy, resInfo *resourceInfo) (*bucketPolicy
 		getAllowStatement(resInfo, acl.Owner.ID, aclFullControl),
 	}
 
+	// Expect to have at least 1 full control grant for owner which is set in
+	// parseACLHeaders(). If there is no other grants, then user sets private
+	// canned ACL, which is processed in this branch.
+	if len(acl.AccessControlList) < 2 {
+		results = append([]statement{getDenyStatement(resInfo, allUsersWildcard, aclFullControl)}, results...)
```

Comment:
Can we add deny records in `addPredefinedACP` ?
https://github.com/nspcc-dev/neofs-s3-gw/blob/7ba7e7dc4db079209593cfa0d7f79b2e722afced/api/handler/acl.go#L581